src/types/models/token-intents.ts

```diff
@@ -1,12 +1,13 @@
 import type {
+  TokenBase,
   DataObject,
-  TokenData,
   Auditable,
   TokenType,
 } from '@/types/models';
 
 interface TokenIntentCardDetails {
-  cardDetails: {
+  type: 'card';
+  card: {
     bin: string;
     last4: string;
     brand: string;
```

Comment:
We're calling `type` below `funding` and it can have values `credit`, `debit`, or `prepaid`.